### PR TITLE
check class of payments are same

### PR DESCRIPTION
**Pull request by jphillips2121**
Added a validate method that checks that all classOfPayments within a transaction are the same and the resource can therefore be paid for. 
Once this has been established the classOfPayment is used to determine the bearer token that is sent to GovPay, and therefore the final destination of the payment. 

Resolves: LFA-446 

### Type of change

* [X] New feature

### Pull request checklist

* [X] I have added unit tests for new code that I have added
* [X] I have added/updated functional tests where appropriate
* [X] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/golang/go/wiki/CodeReviewComments)__

### CODE REVIEW

File: service/external_paysession.go [hunk 1]
```diff
@@ -44,3 +51,21 @@ func (service *PaymentService) CreateExternalPaymentJourney(req *http.Request, p
 		return nil, Error, err
 	}
 }
+
+func validateClassOfPayment(costs *[]models.CostResourceRest) error {
+
+	for i, cost := range *costs {
+		//Loop through Class Of Payments on a single resource to check they're the same.
+		for j, classOfPayment := range cost.ClassOfPayment {
+			if classOfPayment[j] != classOfPayment[0] {
```
Comment (emead) [inline service/external_paysession.go]: Does this work for the following scenario?
```
classOfPayment[0] = 1
classOfPayment[1] = 2
classOfPayment[2] = 2
```

Comment (jphillips2121) [reply service/external_paysession.go]: Yes it checks if there's a difference between the [0] value and every value afterwards. So in this instance as soon as it evaluates classOfPayment[1] != classOfPayment[0] it'll return an error.

File: service/gov_pay.go [hunk 2]
```diff
@@ -61,8 +61,14 @@ func (gp *GovPayService) GenerateNextURLGovPay(req *http.Request, paymentResourc
 		return "", Error, fmt.Errorf("error generating request for GovPay: [%s]", err)
 	}
 
+	if paymentResource.Costs[0].ClassOfPayment[0] == "penalty" {
```
Comment (dalebradley) [inline service/gov_pay.go]: Should be a constant

File: service/gov_pay.go [hunk 3]
```diff
@@ -61,8 +61,14 @@ func (gp *GovPayService) GenerateNextURLGovPay(req *http.Request, paymentResourc
...
+		request.Header.Add("authorization", "Bearer "+gp.PaymentService.Config.GovPayBearerTokenTreasury)
+	}
+	if paymentResource.Costs[0].ClassOfPayment[0] == "data-maintenance" {
```
Comment (dalebradley) [inline service/gov_pay.go]: Should be a constant

File: service/gov_pay.go [hunk 4]
```diff
@@ -61,8 +61,14 @@ func (gp *GovPayService) GenerateNextURLGovPay(req *http.Request, paymentResourc
...
+		request.Header.Add("authorization", "Bearer "+gp.PaymentService.Config.GovPayBearerTokenChAccount)
```
Comment (dalebradley) [inline service/gov_pay.go]: What happens if there is no class of payment? Or one with a different values than specified above? How about a switch statement with a default case?

**Pull request merged**